radio_beam/commonbeam.py

```diff
@@ -0,0 +1,114 @@
+
+import numpy as np
+import astropy.units as u
+
+from .beam import Beam
+from .utils import BeamError, transform_ellipse
+
+
+def commonbeam(beams):
+    '''
+    Find a common beam that can be convolved to from a `Beams` object. This
+    function is based on the CASA implementation `ia.commonbeam`.
+
+    Parameters
+    ----------
+    beams : `~radio_beam.Beams`
+
+    Returns
+    -------
+    common_beam : `~radio_beam.Beam`
+        The smallest common beam in the set of beams.
+    '''
+
+    # This code is based on the implementation in CASA:
+    # https://open-bitbucket.nrao.edu/projects/CASA/repos/casa/browse/code/imageanalysis/ImageAnalysis/CasaImageBeamSet.cc
+
+    if (~beams.isfinite).all():
+        raise BeamError("All beams in the object are invalid.")
+
+    largest_beam = beams.largest_beam()
+    largest_major = largest_beam.major
+    largest_minor = largest_beam.minor
+
+    for beam in beams:
+
+        if beam != largest_beam and beam.isfinite:
+            deconv_beam = \
+                largest_beam.deconvolve(beam, failure_returns_pointlike=True)
+            if not deconv_beam.isfinite:
+
+                pa_diff = (beam.pa - largest_beam.pa).to(u.rad)
+
+                problem_major = beam.major
+                problem_minor = beam.minor
+
+                # If the difference is pi / 2, the larger major is set to the
+                # new major and the minor is the other major.
+                if np.abs(pa_diff).value == np.pi / 2.:
```

Comment:
this will almost certainly never be true because of numerical effects.  Some other equality test (`np.testing.assert_almost_equal`.... I think there's a version that returns a boolean).